hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -437,15 +350,26 @@ public DatanodeDeletedBlockTransactions getTransactions(
       throws IOException {
     lock.lock();
     try {
+      // Here we can clean up the Datanode timeout command that no longer
+      // reports heartbeats
+      getSCMDeletedBlockTransactionStatusManager().cleanAllTimeoutSCMCommand(
```

Comment:
- "SCMBlockDeletingService timeout": is `SCMBlockDeletingService` itself timeout, When the `SCMBlockDeletingService` thread is stuck, it may trigger the

- `scmCommandTimeoutMs`: is the `scmCommand` is the maximum time allowed without an update, is used to prevent a commands from being lost in the process of sending them to the DN, we can resend them after a timeout.
`scmCommand` can be update by DN Heartbeat, If no DN reports the status of a `scmCommand` for 300s, then the `scmCommand` will be cleaned up due to a timeout.

Reply:
@xichen01 
"SCMBlockDeletingService timeout": this is period after which it will create block set for further blocks for deletion. So this is running every 5 min.
scmCommandTimeoutMs: this is waiting time after command is sent in HB for retry. 
So normally DN should execute delete blocks within "5" minute.

With both having "5 min", and scmCommandTimeoutMs being initialized after some time, it will avoid sending same command for one more retry, and effectively get 10 min for blocks deletion.

This is more useful for aggressive deletion, keeping timeout to "1 min" for "SCMBlockDeletingService timeout", so that new blocks are prepared quickly even at background, old blocks deletion in progress at DN. And also resolving duplicate blocks to that extent.

IMO, scmCommandTimeoutMs by default can be set to "20 min" as default value in corresponding to "SCMBlockDeletingService timeout" which is default '5 min", i.e. it will get 4-5 cycle waiting time for sending same blocks again.
OR can have relative configuration as scmCommandTimeoutMs = 4 * "SCMBlockDeletingService timeout"

Reply:
@xichen01 Yes, you are correct. Above is fine.